src/main/java/core/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package core.dao.impl;
+
+import core.dao.CinemaHallDao;
+import core.lib.Dao;
+import core.model.CinemaHall;
+import core.model.exception.DataProcessingException;
+import core.util.HibernateUtils;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class CinemaHallDaoImpl implements CinemaHallDao {
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtils.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(cinemaHall);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null || transaction.isActive()) {
```

Comment:
```suggestion
            if (transaction != null || transaction.isActive()) {
```
I think, we should use here && instead of ||